commons/src/main/java/org/infinispan/commons/dataconversion/MediaType.java

```diff
@@ -0,0 +1,231 @@
+   public static final String APPLICATION_JBOSS_MARSHALLING_TYPE = "application/x-jboss-marshalling";
+   public static final String APPLICATION_PROTOSTREAM_TYPE = "application/x-protostream";
+   public static final String IMAGE_GIF_TYPE = "image/gif";
+   public static final String IMAGE_JPEG_TYPE = "image/jpeg";
+   public static final String IMAGE_PNG_TYPE = "image/png";
+   public static final String TEXT_CSS_TYPE = "text/css";
+   public static final String TEXT_CSV_TYPE = "text/csv";
+   public static final String TEXT_PLAIN_TYPE = "text/plain";
+   public static final String TEXT_HTML_TYPE = "text/html";
+   public static final String APPLICATION_INFINISPAN_MARSHALLING_TYPE = "application/x-infinispan-marshalling";
+   public static final String APPLICATION_INFINISPAN_BINARY_TYPE = "application/x-infinispan-binary";
+   public static final String APPLICATION_PROTOSTUFF_TYPE = "application/x-protostuff";
+   public static final String APPLICATION_KRYO_TYPE = "application/x-kryo";
+
+   public static MediaType APPLICATION_JSON = fromString(APPLICATION_JSON_TYPE);
+   public static MediaType APPLICATION_OCTET_STREAM = fromString(APPLICATION_OCTET_STREAM_TYPE);
+   public static MediaType APPLICATION_OBJECT = fromString(APPLICATION_OBJECT_TYPE);
+   public static MediaType APPLICATION_SERIALIZED_OBJECT = fromString(APPLICATION_SERIALIZED_OBJECT_TYPE);
+   public static MediaType APPLICATION_XML = fromString(APPLICATION_XML_TYPE);
+   public static MediaType APPLICATION_PROTOSTREAM = fromString(APPLICATION_PROTOSTREAM_TYPE);
+   public static MediaType APPLICATION_JBOSS_MARSHALLED = fromString(APPLICATION_JBOSS_MARSHALLING_TYPE);
+   public static MediaType APPLICATION_INFINISPAN_MARSHALLED = fromString(APPLICATION_INFINISPAN_MARSHALLING_TYPE);
+   public static MediaType IMAGE_PNG = fromString(IMAGE_PNG_TYPE);
+   public static MediaType TEXT_PLAIN = fromString(TEXT_PLAIN_TYPE);
+   public static MediaType TEXT_HTML = fromString(TEXT_HTML_TYPE);
+   public static MediaType APPLICATION_PROTOSTUFF = fromString(APPLICATION_PROTOSTUFF_TYPE);
+   public static MediaType APPLICATION_KRYO = fromString(APPLICATION_KRYO_TYPE);
+   public static MediaType APPLICATION_INFINISPAN_BINARY = fromString(APPLICATION_INFINISPAN_BINARY_TYPE);
+
+   private static final String INVALID_TOKENS = "()<>@,;:/[]?=\\\"";
+
+   private final Map<String, String> params = new HashMap<>(2);
+   private final String type;
+   private final String subType;
+   private final String typeSubtype;
+
+   public MediaType(String type, String subtype, Map<String, String> params) {
+      this(type, subtype);
+      params.forEach(this.params::put);
+   }
+
+   public MediaType(String type, String subtype) {
+      this.type = validate(type);
+      this.subType = validate(subtype);
+      this.typeSubtype = type + "/" + subtype;
+   }
+
+   public static MediaType fromString(String mediaType) {
+      return parse(mediaType);
+   }
+
+   public static MediaType parse(String str) {
```

Comment:
You could have used a javax.activation.MimeType which has all of the type, subtype and parameter parsing logic

Reply:
I was using it initially but ditched. It'd involve messing with jboss modules to add it as a dependency, plus I didn't like the fact its constructor throws a checked exception.

Reply:
javax.activation should be part of SE since 7, but +1 on avoiding the checked exception

Reply:
You're right, but this package is not exposed by default in Wildfly, needs explicit dependency